README.md

```diff
@@ -1,21 +1,192 @@
+
+```
+in: {"num": 42}
+template: {"answer": "The answer to the Ultimate Question of Life, the Universe, and Everything is {{ $.num }}"}
+out: {"answer": "The answer to the Ultimate Question of Life, the Universe, and Everything is 42"}
+```
+
+Beware that non scalar values (such as arrays) cannot be converted to string, therefore a rendering error is raised.
+
+String interpolation can be avoided using `{{{ }}}`, which is the preferred method for templating non string values:
+```
+in: {"number": 5}
+t: {"a": "{{{ $.number }}}"}
+out: {"a": 5}
+```
+
+`{{{ }}}` cannot be mixed with string interpolation, therefore `" {{{ $.test }}} a" causes a rendering error.
+
+### `&` (Unquote) Operator
 
-Documentation can be generated with [ExDoc](https://github.com/elixir-lang/ex_doc)
-and published on [HexDocs](https://hexdocs.pm). Once published, the docs can
-be found at [https://hexdocs.pm/exjsontemplate](https://hexdocs.pm/exjsontemplate).
+Unquote operator should be used when a scalar value is represented as string (such as `"5"`):
 
+```
+in: {"number": "5"}
+t: {"a": "{{& $.number}}"}
+out: {"a": 5}
+```
+
+If a string cannot be parsed as a scalar value, a rendering error is returned.
+
+When `&` operator is applied to any non string value, it behaves like `{{{ }}}`:
+
+```
+in: {"number": 5}
+t: {"a": "{{& $.number}}"}
+out: {"a": 5}
+```
+
+`&` can be also used for non scalar values such as:
+
+```
+in: {"nums": [0, 1, 2, 3, 4, 5]}
+t: {"nums2": ["{{& $.nums}}"]}
+out: {"nums2": [[0, 1, 2, 3, 4, 5]]}
+```
+
+`&` never renders strings, therefore it cannot be used for string interpolation.
+
+### `#` (Section) Operator
+
+Array templates can be written using `#` section operator:
+
+```
+in: {"repo": [{"name": "Davide"}, {"name": "Riccardo"}]}
+t: {"{{#repo}}": "Hello {{ $.name }}"}
+out: ["Hello Davide", "Hello Riccardo"]
+```
+
+```
+in: {"repo": ["Davide", "Riccardo"]}
+t: {"{{#repo}}": "Hello {{ $ }}"}
+out: ["Hello Davide", "Hello Riccardo"]
+```
+
+`#` operator can be also used for optional sections when applied to a truthy scalar:
+
+```
+in: {"person": true, "name": "Davide"}
+t: {"{{#person}}": "Hello {{$.name}}"}
+out: "Hello Davide"
+```
+
+### `^` (Inverted Optional Section) Operator
+
+Optional sections can be left out using `^` operator:
+
+```
+in: {"person": true, "name": "Davide"}
+t: {"a": 1, "b": {"{{^person}}": "Hello {{$.name}}"}}
+out: {"a": 1}
+```
+
+When the whole template is optional, `null` is rendered.
+
+```
+in: {"person": true, "name": "Davide"}
+t: {"{{^person}}": "Hello {{$.name}}"}
+out: null
+```
+
+Each optional section can have only a single item.
+
+### `?` (Switch) Operator
+
+Switch operator is useful when handling multiple cases, such as when dealing with enumeration:
+
+```
+in: {"num": 1}
+t: {"message": {"{{?num}}": [{"case": 1, "template": "one"}, {"case": 2, "template": "two"}, {"case": 3, "template": "three"}, {"template": "A lot."}]}}
```

Comment:
This and all the long examples below would probably benefit from multiline + indentation